src/internal.rs

```diff
@@ -25,15 +44,66 @@ use core::sync::atomic::Ordering;
 use alloc::boxed::Box;
 
 use crossbeam_utils::cache_padded::CachePadded;
+use arrayvec::ArrayVec;
 
 use atomic::Owned;
 use collector::{Handle, Collector};
 use epoch::{AtomicEpoch, Epoch};
 use guard::{unprotected, Guard};
-use garbage::{Bag, Garbage};
+use deferred::Deferred;
 use sync::list::{List, Entry, IterError, IsElement};
 use sync::queue::Queue;
 
+/// Maximum number of objects a bag can contain.
+#[cfg(not(feature = "strict_gc"))]
+const MAX_OBJECTS: usize = 64;
+#[cfg(feature = "strict_gc")]
+const MAX_OBJECTS: usize = 4;
+
+/// Bag of garbages.
+#[derive(Default, Debug)]
+pub struct Bag {
+    /// Stashed objects.
+    deferreds: ArrayVec<[Deferred; MAX_OBJECTS]>,
+}
+
+// It is safe to implement `Send` for `Bag`, since the user of `Bag::try_push()` guarantees that
+// another thread may execute any `Deferred` functions inside a `Bag`.
+unsafe impl Send for Bag {}
```

Comment:
Wouldn't it make more sense to `impl Send` for `Deferred` instead? It will then automatically be applied to `Bag`.

Reply:
I'd like to cc @stjepang.  He said that [`Deferred` is better not to implement `Send`](https://github.com/crossbeam-rs/crossbeam-epoch/issues/12#issuecomment-336735548).

Reply:
Actually, I agree with @Amanieu. Let's have `impl Send for Deferred {}`